### DISCUSSION TITLE
Dealing with the cascade in forms whens using sections

### DISCUSSION THREAD

**Discussion by earthwytch** [statamic/cms#8604] [Q&A] [answered]
In a page template that contains a form, the `{{ instructions }}` for a section are being injected into the instructions for the underlying fields if they don't have their own instructions. I can see it is a cascade issue, but I don't know how to fix it. I tried using `{{ field:instructions }}` and `{{ fields:instructions }}` but that didn't work.

```
{{ sections }}
    <fieldset>
        <legend>
            <h3>{{ display }}</h3>
        </legend>
        <p>{{ instructions }}</p>

    <div>
        {{ fields }}
            {{ if visibility == 'hidden' }}
                {{ field }}
            {{ else }}
                <div id="{{ handle | dashify }}">

                    <label for="{{ handle }}">{{ display }}
                        {{ if instructions }}<br />{{ instructions }}{{ /if }}
                    </label>

                    {{ field }}

                </div>
            {{ /if }}
        {{ /fields }}
    </div>

    </fieldset>
{{ /sections }}
```

Any suggestions?

![image](https://github.com/statamic/cms/assets/4753015/dd01eff9-5543-4f80-95fd-819659586a6e)

**Comment by earthwytch** [comment, marked as answer]
In case anyone else ever has this issue, I have dealt with it by setting the section instructions to a variable, and then checking against that variable when it came time to display the field instructions.

```
{{ sections }}
    <fieldset>
        <legend>
            <h3>{{ display }}</h3>
        </legend>
        <p>{{ daddio = instructions }}{{ daddio }}</p>

    <div>
        {{ fields }}
            {{ if visibility == 'hidden' }}
                {{ field }}
            {{ else }}
                <div id="{{ handle | dashify }}">

                    <label for="{{ handle }}">{{ display }}
                        {{ if instructions != daddio }}<br />{{ instructions }}{{ /if }}
                    </label>

                    {{ field }}

                </div>
            {{ /if }}
        {{ /fields }}
    </div>

    </fieldset>
{{ /sections }}
```

Feels kinda hacky but it works! 👍

**Reply by jesseleite** [reply]
I think this is a bug though (or at least an oversight). I'll take a look into it! Feel free to follow along on progress here https://github.com/statamic/cms/issues/8646

**Reply by jesseleite** [reply]
Fixed in https://github.com/statamic/cms/pull/8651, will likely be in next release or two @earthwytch 👍
